tools/loader/ImageClassifier.cpp

```diff
@@ -238,6 +245,51 @@ static void printTopKPairs(const std::vector<FloatIndexPair> &topKPairs) {
   }
 }
 
+/// Convert all placeholders with elements of type \p srcTy into type \p destTy.
+/// Also explicitly convert the input data and the output tensor into the type
+/// \p destTy.
+static void convertPlaceholders(Loader &loader, Context &ctx, Tensor &data,
+                                Placeholder *resultVar, ElemKind srcTy,
+                                ElemKind destTy) {
+  // Convert inputs and outputs of type srcTy to use destTy.
+  for (auto &pair : ctx.pairs()) {
+    auto *PH = pair.first;
+    auto *payload = pair.second;
+    if (PH->getElementType() != srcTy) {
+      continue;
+    }
+    // Create the type that uses destTy.
+    auto convertedTy =
+        loader.getFunction()->getParent()->uniqueType(destTy, PH->dims());
+    PH->setType(0, convertedTy);
+    if (payload) {
+      // Convert the payload of the tensor
+      payload->convertToType(destTy);
+    }
+  }
+  data.convertToType(destTy);
+  for (auto &user : resultVar->getUsers()) {
+    if (auto *SN = llvm::dyn_cast<SaveNode>(user.getUser())) {
+      SN->setNthInput(0, loader.getFunction()->createConvertTo(
+                             "cast", SN->getInput(), resultVar->getType()));
+    }
+  }
+}
```

Comment:
This logic is essentially already in the function converter.
I'll post a PR with the required refactoring. Either wait for that PR and you won't have to cover that part, or I'll update if it lands before.

Reply:
That's in #1905